test/test_framework.py

```diff
@@ -870,6 +876,8 @@ class SomeObject(Object):
             obj.state.a = get_a()
             framework.commit()
 
+            framework._forget(obj)
+            framework._forget(obj.state._data)
```

Comment:
Having to dig into deep internal details this way is quite unfortunate. These tests probably won't make much sense even to ourselves in a few months.

I can think of two alternatives:

1. Create a different framework, so we can look at a copy of the object instead.
2. Rely on garbage collection to clean things up for us.

Reply:
In the meeting I brought up the idea of having `framework._forget` have special logic for `StoredState` and I'm still thinking that might be reasonable.  Having `StoredState` objects will be relatively common and they will need to be explicitly forgotten when the parent object is, so if we assume that people will need to do that, we shouldn't require them to know the implementation detail on how to do so, we should just handle it internally.

Reply:
Doesn't sound sane. We should allow people to build their own hierarchies without having to stumble upon internal details like this. If we can't make it work for ourselves without such hardcoding, we can't expect people to make it work for themselves either.